protobuf/protobuf.go

```diff
@@ -27,13 +28,17 @@ func (fcd FirehoseEncoderDecoder) EncodeData(data interface{}, useMessageTranspo
 	if useMessageTransport {
 		panic("Message Transport should not be used for firehose encoding")
 	}
-	dst := anypb.Any{}
-	dst.Value = data.([]byte)
-	msgType, ver, _ := fcd.DecodeMessageType(metaAttrs.Schema)
-	dst.TypeUrl = fcd.typeUrls[hedwig.MessageTypeMajorVersion{
-		MessageType:  msgType,
-		MajorVersion: uint(ver.Major()),
-	}]
+	var dst *anypb.Any
+	if reflect.TypeOf(data) == reflect.TypeOf([]byte(nil)) {
+		dst.Value = data.([]byte)
+		msgType, ver, _ := fcd.DecodeMessageType(metaAttrs.Schema)
+		dst.TypeUrl = fcd.typeUrls[hedwig.MessageTypeMajorVersion{
+			MessageType:  msgType,
+			MajorVersion: uint(ver.Major()),
+		}]
+	} else {
```

Comment:
a comment would be good here. I'm guessing it's to be able to read firehose file and write it back.